controllers/k8ssandra/medusa_reconciler.go

```diff
@@ -56,13 +76,57 @@ func (r *K8ssandraClusterReconciler) reconcileMedusa(
 			}
 			secret.AddInjectionAnnotationMedusaContainers(&dcConfig.Meta.Pods, cassandraUserSecretName)
 		}
+
+		// Reconcile the Medusa standalone deployment
+		kcKey := utils.GetKey(kc)
+		desiredMedusaStandalone.SetLabels(labels.CleanedUpByLabels(kcKey))
+		recRes := reconciliation.ReconcileObject(ctx, remoteClient, r.DefaultDelay, *desiredMedusaStandalone)
+		switch {
+		case recRes.IsError():
+			return recRes
+		case recRes.IsRequeue():
+			return recRes
+		}
+
+		// Create and reconcile the Medusa service for the standalone deployment
+		medusaService := medusa.StandaloneMedusaService(dcConfig, medusaSpec, kc.Name, namespace, logger)
+		medusaService.SetLabels(labels.CleanedUpByLabels(kcKey))
+		recRes = reconciliation.ReconcileObject(ctx, remoteClient, r.DefaultDelay, *medusaService)
+		switch {
+		case recRes.IsError():
+			return recRes
+		case recRes.IsRequeue():
+			return recRes
+		}
+
+		// Check if the Medusa Standalone deployment is ready, and requeue if not
+		ready, err := r.isMedusaStandaloneReady(ctx, remoteClient, desiredMedusaStandalone)
+		if err != nil {
+			logger.Info("Failed to check if Medusa standalone deployment is ready", "error", err)
+			return result.Error(err)
+		}
+		if !ready {
+			logger.Info("Medusa standalone deployment is not ready yet")
+			return result.RequeueSoon(r.DefaultDelay)
+		}
 	} else {
 		logger.Info("Medusa is not enabled")
 	}
 
 	return result.Continue()
 }
 
+// Check if the Medusa standalone deployment is ready
+func (r *K8ssandraClusterReconciler) isMedusaStandaloneReady(ctx context.Context, remoteClient client.Client, desiredMedusaStandalone *appsv1.Deployment) (bool, error) {
+	// Get the medusa standalone deployment and check the number of ready replicas
+	deplKey := utils.GetKey(desiredMedusaStandalone)
+	medusaStandalone := &appsv1.Deployment{}
+	if err := remoteClient.Get(context.Background(), deplKey, medusaStandalone); err != nil {
+		return false, err
+	}
+	return medusaStandalone.Status.ReadyReplicas > 0, nil
```

Comment:
Issue: I think this should more likely be built around DeploymentCondition: Available == true instead of ReadyReplicas.